marine/marine_pool.py

```diff
@@ -0,0 +1,52 @@
+import itertools
+import math
+import multiprocessing
+from dataclasses import dataclass
+from functools import partial
+from typing import List, Dict, Optional, Tuple
+
+from marine import Marine
+
+
+@dataclass
+class MarinePool:
+    _lib_path: str
+    process_count: int = 4
+
+    def filter_and_parse(
+        self,
+        packets: List[bytes],
+        bpf: Optional[str] = None,
+        display_filter: Optional[str] = None,
+        fields: Optional[List[str]] = None,
+    ) -> List[Tuple[bool, Dict[str, str]]]:
+        if len(packets) == 0:
+            return []
+
+        ctx = multiprocessing.get_context("spawn")
+        pool = ctx.Pool(self.process_count)
+        chunk_size = int(math.ceil(len(packets) / float(self.process_count)))
+        packet_chunks = [
```

Comment:
`multiprocessing.Pool.map` already splits the input to batches for you and exposes a clean API.